### PR TITLE
Adding eks-stig-kubernetes-v1r6

### LINKED ISSUE #1247: DoD STIG Ver 1 Rel 6
#833 

Adding tests to support DoD STIG Ver 1 Rel 6

### PR DESCRIPTION
**Pull request by crenzoaws**
Closes #1247

### CODE REVIEW

**Comment by mozillazg** [comment]
@crenzoaws Thanks for your contribution! I'll review it when I have free time. Would you please update the document at https://github.com/aquasecurity/kube-bench/blob/main/docs/platforms.md to include this new benchmark?

**Comment by crenzoaws** [comment]
@mozillazg 
- The update to https://github.com/aquasecurity/kube-bench/blob/main/docs/platforms.md is included. 
- Submitted another commit to fix the lint errors identified in the previous check

**Comment by mozillazg** [comment]
@crenzoaws Thanks for your update. I'll review it later.

File: hack/kind-stig.test.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+spec:
+  template:
+    metadata:
+      labels:
+        app: kube-bench
+    spec:
+      hostPID: true
+      containers:
+        - name: kube-bench
+          image: docker.io/aquasec/kube-bench:5c76890
```
Comment (mozillazg) [inline hack/kind-stig.test.yaml:15]:
```suggestion
          image: docker.io/aquasec/kube-bench:latest
```

**Review by mozillazg — CHANGES_REQUESTED** [review]
Thanks for your contribution! Added some comments please check again, thanks!

File: hack/kind-stig.test.yaml (hunk 1 above)
Comment (crenzoaws) [reply hack/kind-stig.test.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        tests:
+          test_items:
+            - flag: root:root
+        remediation: |
+          Run the below command (based on the file location on your system) on the each worker node.
+          For example,
+          chown root:root $kubeletkubeconfig
+        scored: true
+      - id: V-242407
+        text: "Ensure that the kubelet configuration file has permissions set to 644 or more restrictive (Automated)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        remediation: |
+          Edit the kubelet service file $kubeletbin
+          on each worker node and remove the --hostname-override argument from the
+          KUBELET_SYSTEM_PODS_ARGS variable.
+          Based on your system, restart the kubelet service. For example:
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+        scored: true
+      - id: V-242406
+        text: "Ensure that the kubelet kubeconfig file ownership is set to root:root (Automated)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+              set: false
+        remediation: |
+          Edit any manifest files or $kubeletconf that contain the feature-gates
+          setting with AllAlpha set to "true".
+          Set the flag to "false" or remove the "AllAlpha" setting
+          completely. Restart the kubelet service if the kubelet config file
+          if the kubelet config file is changed.
+        scored: true
+      - id: V-242404  # CIS 3.2.8
+        text: "Ensure that the --hostname-override argument is not set (Automated)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+          If using command line arguments, edit the kubelet service file
+          $kubeletsvc on each worker node and
+          set the below parameter in KUBELET_SYSTEM_PODS_ARGS variable.
+          --protect-kernel-defaults=true
+          Based on your system, restart the kubelet service. For example:
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+        scored: true
+      - id: V-242435
+        text: "Ensure that the --authorization-mode argument is not set to AlwaysAllow (Automated)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+          If using executable arguments, edit the kubelet service file
+          $kubeletsvc on each worker node and
+          set the below parameter in KUBELET_SYSTEM_PODS_ARGS variable.
+          --anonymous-auth=false
+          Based on your system, restart the kubelet service. For example:
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+        scored: true
+      - id: V-242392  # CIS 3.2.2
+        text: "Ensure that the --authorization-mode argument is not set to AlwaysAllow (Automated)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+        type: "manual"
+        remediation: |
+         Run the following command:
+         kubectl get all -o jsonpath='{range .items[?(@..secretKeyRef)]} {.kind} {.metadata.name} {"\n"}{end}' -A
+         If any of the values returned reference environment variables
+         rewrite application code to read secrets from mounted secret files, rather than
+         from environment variables.
+        scored: false
+      - id: V-242434  # CIS 3.2.6
+        text: "Ensure that the --protect-kernel-defaults argument is set to true (Automated)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+          If using command line arguments, edit the kubelet service file
+          $kubeletsvc on each worker node and
+          set the below parameter in KUBELET_SYSTEM_PODS_ARGS variable.
+          --read-only-port=0
+          Based on your system, restart the kubelet service. For example:
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+        scored: false
+      - id: V-242390
+        text: "Ensure that the --anonymous-auth argument is set to false (Automated)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Update test and moved to controlplane.yaml along with test - V-242400 The Kubernetes API server must have Alpha APIs disabled (Automated)

File: cfg/eks-stig-kubernetes-v1r6/node.yaml [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+version: "eks-stig-kubernetes-v1r6"
+id: 3
+text: "Worker Node Security Configuration"
+type: "node"
+groups:
+  - id: 3.1
+    text: "DISA Category Code I"
+    checks:
+      - id: V-242387   # CIS 3.2.4
+        text: "Ensure that the --read-only-port argument is set to 0 (Manual)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/node.yaml]: Updated

File: cfg/eks-stig-kubernetes-v1r6/controlplane.yaml [hunk 10]
```diff
@@ -0,0 +1,77 @@
+---
+controls:
+version: "eks-stig-kubernetes-v1r6"
+id: 2
+text: "Control Plane Configuration"
+type: "controlplane"
+groups:
+  - id: 2.2
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/controlplane.yaml:55]: As a result of moving other tests to controlplane.yaml a group 2.1 was created to represent DISA Category Code I tests. This was left as is to represent DISA Category Code II tests.

File: cfg/eks-stig-kubernetes-v1r6/controlplane.yaml [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        text: "The Kubernetes Controller Manager must create unique service accounts for each work payload. (Manual)"
+        type: "manual"
+        remediation: |
+          Create explicit service accounts wherever a Kubernetes workload requires specific access
+          to the Kubernetes API server.
+          Modify the configuration of each default service account to include this value
+          automountServiceAccountToken: false
+        scored: false
+      - id: V-242402
+        text: "Enable audit logs (Manual)"
```
[the remark this answers is not in the document]
Comment (crenzoaws) [reply cfg/eks-stig-kubernetes-v1r6/controlplane.yaml]: Updated

**Comment by crenzoaws** [comment]
@mozillazg - Thanks for all your feedback. I've incorporated all the changes you've identified.

**Comment by crenzoaws** [comment]
Upstream dependency failure:  sgerrand/alpine-pkg-glibc#185

**Comment by crenzoaws** [comment]
Upstream package error: sgerrand/alpine-pkg-glibc#185

**Comment by mozillazg** [comment]
> @mozillazg - Thanks for all your feedback. I've incorporated all the changes you've identified.

Thanks for your patience. It looks like the `integration/testdata/Expected_output_stig.data` file needed to be updated too.

**Review by mozillazg — APPROVED** [review]
LGTM, Thanks for your contribution!

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**